### PR TITLE
Adds authentication CASE-992

### PR DESCRIPTION
**Pull request by edgibbs**
So this runs locally using the containers in integrated-test-environment.  If you run a local rails server you should get redirected to perry and a login.  Apparently, perry takes anything so it will then redirect you back with a valid token that is stored in the session.  Currently, the session is cookie based, so a small story will be too hook it up to the redis store instead.  

This mostly uses a rack middleware class that intercepts all requests and does this check.  I figured it makes the most sense and it could be gemified more easily.  What I still am unsure about is how they're passing environment variables like the RAILS_ENV into the docker containers, but I'm going to try this out in preint this afternoon.

### CODE REVIEW

**Comment by edgibbs** [comment]
Oh, I forgot I added the feature gem as well and defaulted it to ignoring these checks since I didn't want to break other people.

File: lib/infrastructure/cwds_authenticator.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+module Infrastructure
+  class CwdsAuthenticator
+    def initialize(application)
+      @application = application
+    end
+
+    def call(environment)
+      return @application.call(environment) unless Feature.active?(:authentication)
+      request = Rack::Request.new(environment)
+      return redirect_to_login(request.url) if no_token_or_session(request)
```
Comment (eckdanny-osi) [inline lib/infrastructure/cwds_authenticator.rb]: Q: does this apply to ajax requests, too? As a consumer I'd expect an HTTP 401 for `/api/*` reqs.

File: config/micro_services.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 development:
   case_api_base_url: 'https://casemgmapi.dev.cwds.io'
+  perry_api_base_url: 'http://perry.local.cwds.io'
 test:
   case_api_base_url: 'https://casemgmapi.test.cwds.io'
+  perry_api_base_url: 'https://perry.test.cwds.io'
+preint:
```
Comment (eckdanny-osi) [inline config/micro_services.yml]: @edgibbs mentioned wanting to make a change here. Don't merge until he's done so.

**Comment by eckdanny-osi** [comment]
abused power to merge. Cyclomatic complexity checks rang the rubocop bell. We're okay with that for now.

**Pull request merged**